nncf/experimental/common/check_feature.py

```diff
@@ -12,10 +12,13 @@
 import os
 
 
-def is_experimental_torch_tracing_enabled() -> bool:
+def is_torch_tracing_by_torch_function_mode() -> bool:
     """
-    Checks if experimental torch tracing is enabled by environment variable NNCF_EXPERIMENTAL_TORCH_TRACING.
+    Checks if legacy torch tracing is enabled by environment variable NNCF_TORCH_LEGACY_TRACING.
 
-    :return: True if experimental torch tracing is enabled, False otherwise.
+    True - will wrap model by NNCFNetwork and patch function in torch namespace.
+    False - will use FunctionHookMode without patching torch namespace.
+
+    :return: True if legacy torch tracing is enabled, False otherwise.
     """
-    return os.getenv("NNCF_EXPERIMENTAL_TORCH_TRACING") is not None
+    return os.getenv("NNCF_TORCH_LEGACY_TRACING", "").lower() not in ["1", "on", "true"]
```

Comment:
The user can call patch_torch_operators in their code to enable the tracing via patching. NNCF should handle this for legacy algorithms. At least, `is_torch_tracing_by_torch_function_mode` should handle the case when the tracing via patching is enabled. Regarding the `NNCF_TORCH_LEGACY_TRACING` environment variable, can you provide a user scenario how an user would use it? If there are no custom scripts I would suggest deleting this.